synapse/logging/opentracing.py

```diff
@@ -991,11 +991,7 @@ def _decorator(func: Callable[P, R]) -> Callable[P, R]:
         if not opentracing:
             return func
 
-        # type-ignore: mypy seems to be confused by the ParamSpecs here.
-        # I think the problem is https://github.com/python/mypy/issues/12909
```

Comment:
Aha, nice to see this get fixed up.

Reply:
Oh, is the point that [`strict_concatenate`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-strict_concatenate) was getting in the way of this?

Reply:
Errr, I got a warning saying it was deprecated (and replaced with `--extra-checks` -- that doesn't seem settable on the command line). But I think you're right.